core/node/node_framework/src/implementations/layers/web3_api/tree_api_client.rs

```diff
@@ -10,6 +10,10 @@ use crate::{
     wiring_layer::{WiringError, WiringLayer},
 };
 
+/// Layer that inserts the `TreeApiHttpClient` into the `ServiceContext` resources, if there is no
```

Comment:
AFAIU the HTTP client is currently used as a "fallback" (sort of), so I decided to embed this logic into layer directly, rather than dealing with `resource already exists` outside.